tests/test.rs

```diff
@@ -454,3 +454,18 @@ fn triple_nested_structure() {
 
     assert_eq!(expected_values, nested_objects);
 }
+
+#[test]
+fn uncopyable_iterator() {
+    struct Foo {};
+    impl Iterator for Foo {
+        type Item = ();
+        fn next(&mut self) -> Option<()> {
+            None
+        }
+    }
+    let _ = comp! {
```

Comment:
This is kinda nit picky, put since the original python syntax uses regular parentheses `()` i prefer code examples in this repo use `comp!(...)` syntax

Reply:
One _could_ argue the same way for `[]` as well though. Perhaps I'll put this all on one line to make it not look so weird (to me at least).

Reply:
Yeah, i think one-lining this looks better as well.

You may also like: #6